src/main_controller.ts

```diff
@@ -227,13 +218,7 @@ export class MainController implements vscode.Disposable {
         }
         if (method === "vscode-neovim") {
             const [command, args] = events;
-            extensionCommandManagers.forEach((m) => {
-                try {
-                    m.handleExtensionRequest(command, args);
-                } catch (e) {
-                    logger.error(`${command} failed, args: ${JSON.stringify(args)} error: ${(e as Error).message}`);
-                }
-            });
+            eventBus.fire(command as any, args);
```

Comment:
Definitely looks like some actual simplifications were gained.

P.S. I don't see any logging in eventBus, should there be (e.g. on errors)?

Reply:
I have also thought about this question. I think errors should be handled in the handler rather than the event bus.